sktime/forecasting/compose/_bagging.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file).
+"""Implements Bagging Forecaster."""
+
+__author__ = ["ltsaprounis"]
+
+from copy import deepcopy
+from typing import List, Union
+
+import numpy as np
+import pandas as pd
+from sklearn import clone
+from sklearn.utils import check_random_state
+from sklearn.utils._testing import set_random_state
+
+from sktime.datatypes._utilities import update_data
+from sktime.forecasting.base import BaseForecaster
+from sktime.forecasting.ets import AutoETS
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.bootstrap import STLBootstrapTransformer
+
+
+class BaggingForecaster(BaseForecaster):
+    """Forecast a time series by aggregating forecasts from its bootstraps.
+
+    Bagged "Bootstrap Aggregating" Forecasts are obtained by forecasting bootstrapped
+    time series and then aggregating the resulting forecasts. For the point forecast,
+    the different forecasts are aggregated using the mean function [1]. Prediction
```

Comment:
users might appreciate having a choice here, e.g., an `agg_fun` param. Though we do not need to add it in this PR.